### PR TITLE
release notes 2.0

### PR DESCRIPTION
**Pull request by mr-c**

### CODE REVIEW

**Comment by mr-c** [comment]
@ctb @luizirber @camillescott ready for initial review

**Comment by mr-c** [comment]
- [ ] copy changes over to doc/whats-new-2.0.rst

**Comment by mr-c** [comment]
- [ ] sort entries by relative importance
- [ ] add lines of code change: Python: +2380 / C++: -283

**Comment by ctb** [comment]
On Wed, Sep 02, 2015 at 04:44:30PM -0700, Michael R. Crusoe wrote:
> - [ ] sort entries by relative importance
> - [ ] add lines of code change: Python: +2380 / C++: -283 

# of commits would be interesting too!

**Comment by mr-c** [comment]
1015 commits (!!) so far :-)

**Comment by mr-c** [comment]
(there are 4,904 commits total, so about 1/5 of the commits)

**Comment by ctb** [comment]
On Wed, Sep 02, 2015 at 04:59:42PM -0700, Michael R. Crusoe wrote:
> (there are 4,904 commits total, so about 1/5 of the commits)

nice ;)

File: doc/release-notes/release-2.0.md [hunk 1]
```diff
@@ -0,0 +1,339 @@
+# khmer v2.0 release notes
+
+This is the v2.0 release of khmer featuring and the first from our new lab at
```
Comment (luizirber) [inline doc/release-notes/release-2.0.md]: 'featuring' what?

File: doc/release-notes/release-2.0.md [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+
+## New items of note:
+
+New behavior
+------------
+
+### Streaming I/O from Unix Pipes
+
+All scripts now accept input from named (like `/dev/stdin`, or that
+created using `<( list )` process substituion) and unamed pipes (like
```
Comment (luizirber) [inline doc/release-notes/release-2.0.md]: unnamed

File: doc/release-notes/release-2.0.md [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+\#1055 #1050 #1214 #1179 #1133 #1145 @ctb @qingpeng @bocajnotnef  
+
+### Digital normalization script now supports mixed paired and unpaired read input
+
+`normalize-by-median.py` now supports mixed paired and unpaired (or
+"broken-paired") input. Behavior can be forced to either treat all reads
+as singletons or to require all reads be properly paired
+using `--force\_single` or `--paired`, respectively. If `--paired` is
+set, `--unpaired-reads` can be used to include a file of unpaired reads. The
+`unpaired reads will be examined after all of the other sequence files.
```
Comment (luizirber) [inline doc/release-notes/release-2.0.md]: I think there is an extra ` in the beginning of this line

File: doc/release-notes/release-2.0.md [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+`--force` option #1168 @bocajnotnef
+
+A bug leading to an infinite loop with large gzipped countgraphs was found
+\#1038 #1043 @kdmurray91
+
+All scripts that create nodegraphs or countgraphs report the total number of
+unique k-mers. #491 #609 #429 @mr-c
+
+Read pairs from SRA are fully supported. Reported by @macmanes in #1027, fixed
+by @kdmurray91 @Sherine Awad in #1173 #1088
```
Comment (luizirber) [inline doc/release-notes/release-2.0.md]: Join "Sherine" and "Awad"

File: doc/release-notes/release-2.0.md [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+noisy tests were silenced #1125 #1137 @bocajnotnef
+
+The `check_space()` calls were cleaned up. #1167 #1166 #1170 #993 
+
+Developer docs have been expanded #737 #1184 @bocajnotnef #1083 #1282 @ctb
+@mr-c #1269
+
+A lot of code was deleted: TRACE related code in #274 #1180 @ctb 
+`hashtable_collect_high_abundance_kmers` in #1142 #1044 @ctb `lib/ht-diff.cc`,
+`lib/test-HashTables.cc`, `lib/test-Parser.cc` #1144, @mr-c
```
Comment (luizirber) [inline doc/release-notes/release-2.0.md]: Add the code removed on the license PR? (graphtest, primes, etc?)

File: doc/release-notes/release-2.0.md [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+that use khmer have been added #1063 #1222 @mr-c The help text from the scripts
+has also had a thorough cleanup for formatting. #1268 @mr-c
+
+`fastq-to-fasta.py`'s `--n_keep` option has incorrect help text. We now point
+out that all reads with Ns will be dropped by default unless this option is
+supplied. #657 #814 #1208 @ACharbonneau @bocajnotnef
+
+We've updated the URL to the '88m-reads.fa.gz' file. #1242 #1269 @mr-c
+
+@camillecodon designed and implemented an optimization for
```
Comment (luizirber) [inline doc/release-notes/release-2.0.md]: camillescott

File: doc/release-notes/release-2.0.rst [hunk 7]
```diff
@@ -0,0 +1,385 @@
...
+text from the scripts has also had a thorough cleanup for formatting.
+#1268 @mr-c
+
+``fastq-to-fasta.py``'s ``--n_keep`` option has incorrect help text. We
+now point out that all reads with Ns will be dropped by default unless
+this option is supplied. #657 #814 #1208 @ACharbonneau @bocajnotnef
+
+We've updated the URL to the '88m-reads.fa.gz' file. #1242 #1269 @mr-c
+
+@camillecodon designed and implemented an optimization for
```
Comment (luizirber) [inline doc/release-notes/release-2.0.rst]: camillescott

**Comment by mr-c** [comment]
- [ ] Is it mergeable?
- [ ] Did it pass the tests?
- [ ] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage with `make clean diff-cover`
- [ ] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Is it documented in the ChangeLog?
  http://en.wikipedia.org/wiki/Changelog#Format
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?
- [ ] Do the changes respect streaming IO? (Are they
  tested for streaming IO?)
- [ ] Is the Copyright year up to date?

**Comment by mr-c** [comment]
@luizirber can I get a merge? :-)

**Comment by luizirber** [comment]
LGTM

**Pull request merged**